vslib/src/MACsecFilterStateGuard.cpp

```diff
@@ -0,0 +1,23 @@
+#include "MACsecFilterStateGuard.h"
+
+#include <swss/logger.h>
+
+using namespace saivs;
+
+MACsecFilterStateGuard::MACsecFilterStateGuard(
+    _Inout_ volatile MACsecFilter::MACsecFilterState &guarded_state,
```

Comment:
seems like a overkill here, did you hit scenario that this was optimized in wrong way by compiler that you need volatile here?